### ISSUE TITLE
[Leia] My Subscriptions are not loaded when the GUI language set to non-English

### ISSUE THREAD

**Issue by dtechsrv** [anxdpanic/plugin.video.youtube#754]
* Add-on Version: 7.0.6.3+leia.1
* Kodi Version: 18.9
* Kodi GUI Language: Hungarian
* Operating System: LibreELEC-AML v9.2.8.14 (2024-04-06)
* Operating System Language: English

------

### Current Behavior

It does not load "My Subscriptions" list and indicates that the error has been logged.

But, if the language of the Kodi is set to English, the error will not occur and it loads the Subscriptions list properly.

------

### Log

```
2024-05-14 13:30:29.027 T:3480724352  NOTICE: [plugin.video.youtube] Running: YouTube (7.0.6.3+leia.1) on Leia (Kodi-18.9) with Python 2.7.16
                                            Path: /special/new_uploaded_videos_tv/
                                            Params: {'category_label': 'Feliratkoz\xc3\xa1sok'}
2024-05-14 13:30:29.055 T:3480724352   ERROR: CSetting: error reading <control> tag of "youtube.view.label.color.viewCount"
2024-05-14 13:30:29.055 T:3480724352 WARNING: CSettingGroup: unable to read setting "youtube.view.label.color.viewCount"
2024-05-14 13:30:29.055 T:3480724352   ERROR: CSetting: error reading <control> tag of "youtube.view.label.color.likeCount"
2024-05-14 13:30:29.055 T:3480724352 WARNING: CSettingGroup: unable to read setting "youtube.view.label.color.likeCount"
2024-05-14 13:30:29.056 T:3480724352   ERROR: CSetting: error reading <control> tag of "youtube.view.label.color.commentCount"
2024-05-14 13:30:29.056 T:3480724352 WARNING: CSettingGroup: unable to read setting "youtube.view.label.color.commentCount"
2024-05-14 13:30:41.174 T:3480724352   ERROR: CSetting: error reading <control> tag of "youtube.view.label.color.viewCount"
2024-05-14 13:30:41.174 T:3480724352 WARNING: CSettingGroup: unable to read setting "youtube.view.label.color.viewCount"
2024-05-14 13:30:41.175 T:3480724352   ERROR: CSetting: error reading <control> tag of "youtube.view.label.color.likeCount"
2024-05-14 13:30:41.175 T:3480724352 WARNING: CSettingGroup: unable to read setting "youtube.view.label.color.likeCount"
2024-05-14 13:30:41.175 T:3480724352   ERROR: CSetting: error reading <control> tag of "youtube.view.label.color.commentCount"
2024-05-14 13:30:41.175 T:3480724352 WARNING: CSettingGroup: unable to read setting "youtube.view.label.color.commentCount"
2024-05-14 13:30:41.293 T:3480724352   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.UnicodeEncodeError'>
                                            Error Contents: 'ascii' codec can't encode character u'\xf3' in position 4: ordinal not in range(128)
                                            Traceback (most recent call last):
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/plugin.py", line 17, in <module>
                                                plugin_runner.run(youtube.Provider())
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/plugin_runner.py", line 61, in run
                                                plugin.run(provider, context)
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/plugin/xbmc/xbmc_plugin.py", line 145, in run
                                                for item in result
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/items/xbmc/xbmc_items.py", line 592, in video_listitem
                                                context, local_datetime
                                            UnicodeEncodeError: 'ascii' codec can't encode character u'\xf3' in position 4: ordinal not in range(128)
                                            -->End of Python script error report<--
2024-05-14 13:30:41.613 T:3480724352 WARNING: CPythonInvoker(8, /storage/.kodi/addons/plugin.video.youtube/resources/lib/plugin.py): the python script "/storage/.kodi/addons/plugin.video.youtube/resources/lib/plugin.py" has left several classes in memory that we couldn't clean up. The classes include: N9XBMCAddon9xbmcaddon5AddonE
2024-05-14 13:30:41.631 T:3756614528   ERROR: GetDirectory - Error getting plugin://plugin.video.youtube/special/new_uploaded_videos_tv/?category_label=Feliratkoz%c3%a1sok
2024-05-14 13:30:41.636 T:4091646016   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.youtube/special/new_uploaded_videos_tv/?category_label=Feliratkoz%c3%a1sok) failed
```

------

### Additional Information

This bug only affects the Python2-based version running on Leia.

I first experienced this issue in my own native language (Hungarian), but I also tried a few other languages with the same experience: German, French, Greek, Polish.

**Comment by dtechsrv** [comment]
Meanwhile, I also tested with Kodi 18.9 on Windows and experienced the same result.

```
2024-05-14 15:30:50.215 T:13200  NOTICE: [plugin.video.youtube] Running: YouTube (7.0.6.3+leia.1) on Leia (Kodi-18.9) with Python 2.7.13
                                            Path: /special/new_uploaded_videos_tv/
                                            Params: {'category_label': 'Feliratkoz\xc3\xa1sok'}
2024-05-14 15:30:50.230 T:13200   ERROR: CSetting: error reading <control> tag of "youtube.view.label.color.viewCount"
2024-05-14 15:30:50.230 T:13200 WARNING: CSettingGroup: unable to read setting "youtube.view.label.color.viewCount"
2024-05-14 15:30:50.230 T:13200   ERROR: CSetting: error reading <control> tag of "youtube.view.label.color.likeCount"
2024-05-14 15:30:50.230 T:13200 WARNING: CSettingGroup: unable to read setting "youtube.view.label.color.likeCount"
2024-05-14 15:30:50.230 T:13200   ERROR: CSetting: error reading <control> tag of "youtube.view.label.color.commentCount"
2024-05-14 15:30:50.230 T:13200 WARNING: CSettingGroup: unable to read setting "youtube.view.label.color.commentCount"
2024-05-14 15:30:50.400 T:13200   ERROR: CSetting: error reading <control> tag of "youtube.view.label.color.viewCount"
2024-05-14 15:30:50.400 T:13200 WARNING: CSettingGroup: unable to read setting "youtube.view.label.color.viewCount"
2024-05-14 15:30:50.400 T:13200   ERROR: CSetting: error reading <control> tag of "youtube.view.label.color.likeCount"
2024-05-14 15:30:50.400 T:13200 WARNING: CSettingGroup: unable to read setting "youtube.view.label.color.likeCount"
2024-05-14 15:30:50.400 T:13200   ERROR: CSetting: error reading <control> tag of "youtube.view.label.color.commentCount"
2024-05-14 15:30:50.400 T:13200 WARNING: CSettingGroup: unable to read setting "youtube.view.label.color.commentCount"
2024-05-14 15:30:50.404 T:13200   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.UnicodeEncodeError'>
                                            Error Contents: 'ascii' codec can't encode character u'\xf3' in position 4: ordinal not in range(128)
                                            Traceback (most recent call last):
                                              File "C:\Users\dtech\AppData\Roaming\Kodi\addons\plugin.video.youtube\resources\lib\plugin.py", line 17, in <module>
                                                plugin_runner.run(youtube.Provider())
                                              File "C:\Users\dtech\AppData\Roaming\Kodi\addons\plugin.video.youtube\resources\lib\youtube_plugin\kodion\plugin_runner.py", line 61, in run
                                                plugin.run(provider, context)
                                              File "C:\Users\dtech\AppData\Roaming\Kodi\addons\plugin.video.youtube\resources\lib\youtube_plugin\kodion\plugin\xbmc\xbmc_plugin.py", line 145, in run
                                                for item in result
                                              File "C:\Users\dtech\AppData\Roaming\Kodi\addons\plugin.video.youtube\resources\lib\youtube_plugin\kodion\items\xbmc\xbmc_items.py", line 592, in video_listitem
                                                context, local_datetime
                                            UnicodeEncodeError: 'ascii' codec can't encode character u'\xf3' in position 4: ordinal not in range(128)
                                            -->End of Python script error report<--
```

**Comment by MoojMidge** [comment]
See if this change fixes the problem
https://github.com/MoojMidge/plugin.video.youtube/commit/7c114e0e91362ee076a1b01806370d9ad38ad1d5

**Comment by dtechsrv** [comment]
After changing these lines, the indicated issue is gone.

Thank you! 👍

**Comment by pitsi** [comment]
Thank you for your work!

I tried patching the file myself (on le 9.2.6 x64 for now) and the lines that need to be changed are not 646 and 650 but 587 and 591. In fact, the file is only 647 lines long
```
# cat .kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/items/xbmc/xbmc_items.py | wc -l
647
```
![2024-05-14-201215_1280x1024_scrot](https://github.com/anxdpanic/plugin.video.youtube/assets/5880037/2ee29e15-99e6-48bd-ae2c-009b09f658d4)

**Comment by MoojMidge** [comment]
That is because the linked commit is from a branch where other changes have been made to that file. Those look like the correct lines to change, but in any case here is a patch file:

```
diff --git a/resources/lib/youtube_plugin/kodion/items/xbmc/xbmc_items.py b/resources/lib/youtube_plugin/kodion/items/xbmc/xbmc_items.py
index 0d1f6ed8..3fc94d78 100644
--- a/resources/lib/youtube_plugin/kodion/items/xbmc/xbmc_items.py
+++ b/resources/lib/youtube_plugin/kodion/items/xbmc/xbmc_items.py
@@ -12,7 +12,7 @@ from __future__ import absolute_import, division, unicode_literals
 
 from .. import AudioItem, DirectoryItem, ImageItem, UriItem, VideoItem
 from ...constants import SWITCH_PLAYER_FLAG
-from ...compatibility import xbmc, xbmcgui
+from ...compatibility import to_str, xbmc, xbmcgui
 from ...utils import current_system_version, datetime_parser
 
 
@@ -584,11 +584,11 @@ def video_listitem(context, video_item, show_fanart=None):
     local_datetime = None
     if datetime:
         local_datetime = datetime_parser.utc_to_local(datetime)
-        props['PublishedLocal'] = str(local_datetime)
+        props['PublishedLocal'] = to_str(local_datetime)
     if video_item.live:
         props['PublishedSince'] = context.localize('live')
     elif local_datetime:
-        props['PublishedSince'] = str(datetime_parser.datetime_to_since(
+        props['PublishedSince'] = to_str(datetime_parser.datetime_to_since(
             context, local_datetime
         ))
 
```

**Comment by pitsi** [comment]
Although I will probably open a new issue report later today, does anyone else have their log flooded with messages like these after a search or when selecting a channel etc?
```
2024-05-14 15:30:50.230 T:13200   ERROR: CSetting: error reading <control> tag of "youtube.view.label.color.viewCount"
2024-05-14 15:30:50.230 T:13200 WARNING: CSettingGroup: unable to read setting "youtube.view.label.color.viewCount"
```
Because I do and it makes the addon significantly slower than usual.

**Comment by pitsi** [comment]
I have an addon which uses the youtube addon internally and it fails because of the following. As you can see, the reason is some non-ascii character and it causes a domino effect on 9 files used by the youtube addon (and 2 more on my addon, not mentioned below). What else do I have to change from `str` to `to_srt`?
```
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_resolver.py", line 14, in <module>
                                                from youtube_plugin.youtube.provider import Provider
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/__init__.py", line 13, in <module>
                                                from .provider import Provider
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/provider.py", line 17, in <module>
                                                from .client import YouTube
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/client/__init__.py", line 13, in <module>
                                                from .youtube import YouTube
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/client/youtube.py", line 20, in <module>
                                                from .login_client import LoginClient
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/client/login_client.py", line 15, in <module>
                                                from .__config__ import (
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/client/__config__.py", line 195, in <module>
                                                _api_check = APICheck(XbmcContext())
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/context/xbmc/xbmc_context.py", line 321, in __init__
                                                self.parse_params(dict(parse_qsl(params)))
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/context/abstract_context.py", line 297, in parse_params
                                                param, value
                                            UnicodeDecodeError: 'ascii' codec can't decode byte 0xce in position 0: ordinal not in range(128)
                                            -->End of Python script error report<--
```

**Comment by MoojMidge** [comment]
As I advised you previously:

> If you have an issue follow the issue template and post a complete debug log.

If I had magic powers to be able to know what caused that error with no information, I could think of a bunch of other things I would use them for rather than trying to support a version of this plugin on Kodi 18.

As mentioned before, it is unsupported and you are doing yourself no favours by providing as little information as possible about this error, in an unrelated issue.

I may look into the error, but definitely won't based on what you have provided.

**Comment by pitsi** [comment]
A debug log will reveal all sorts of info (e.g. I have seen my public ip in a libreelec log) and the other addon's name.
I am sorry but no, I'd rather find it myself :)

**Comment by MoojMidge** [comment]
Suit yourself. If you manage to find out anything, feel free to share it here so others can benefit too.

**Comment by pitsi** [comment]
I will, but do not get your hopes up that I will fix it, because I do lack a lot of knowledge.

Ontopic now, have you merged the forementioned commit in 7.0.7+betaX?

**Comment by pitsi** [comment]
Small honest question about python, because my knowledge on it is very limited. When the log says something like that
```
error in file1.py, line 1
  contents of line 1 failing here, pointing to file2.py
error in file2.py, line 2
  contents of line 2 failing here, pointing to file3.py
error in file3.py, line 3
  contents of line 3 failing here, pointing to file4.py
(and so on)
error in file10.py, line 10
  contents of line 10 failing here, and the reason
```
doesn't it mean that the whole... "dominoe" of errors started from line 10 of file10.py and spreaded "upwards", creating the rest? Sorry for asking it publically, but I got fed up searching on lines upon lines of code I can't fully understand for something so trivial.

**Comment by MoojMidge** [comment]
> doesn't it mean that the whole... "dominoe" of errors started from line 10 of file10.py and spreaded "upwards", creating the rest?

Yes. https://realpython.com/python-traceback/

> Sorry for asking it publically, but I got fed up searching on lines upon lines of code I can't fully understand for something so trivial.

I have already advised you that spending too much time on stuff like this will simply mean that no future effort will go towards providing Kodi 18 releases. I don't have time to respond to questions like this to help you learn Python, because you won't provide a log.

**Comment by pitsi** [comment]
Thank you. I asked a friend who is a programmer but does not know much about python and he said that the first line and file is where the dominoe error starts. I guess this does not apply to python. 
Also, the issue does start from youtube as I pointed out above and it is related to encoding. At least I have a starting point.

As for the why I am still fighting with python2, besides having a tvbox which runs the le 9.2 aml build the ts creates.
The main system I watch kodi on is an old laptop running le 9.2 x64 and kodi 18.x. That le version and its kodi on it work with 0 issues. Le 10 and its kodi 19 have some playback issues, even on 480p and lower, so I tested it but never upgraded to it. Le 11 and 12 and their respective kodi versions (20 and 21) have wireless connectivity issues, which is a bigger issue and the main one why I have not upgraded to any of them too.

Other than that, allow me a small rant for the addon that fails to launch. It's dev says it is supported to this day, but it sees one insignificant update once a year and he does not answer on bug reports. And no, commits that change stuff that has no functional impact on the addon, like translations, do not count as development or as "recent activity". 
All this is for his main addon, which he advertises everywhere. He also has a few lesser known ones that are broken and have not seen an update for years.
That is why I'd rather learn python and fix his addons myself than waste my time with issue reports and wait for him for monts or years to reply. Moreover, this way I do not have to deal with the fanboys that find his work "flawless" only because his addon is one of a kind, but have never reported a single issue, although they do know that e.g. some menus don't work.

Thank you for listening.

**Comment by MoojMidge** [comment]
> Thank you. I asked a friend who is a programmer but does not know much about python and he said that the first line and file is where the dominoe error starts. I guess this does not apply to python.

It varies by language, but you are mixing up a few things.

You will see something like this near the start of the Python stack trace `Traceback (most recent call last):` which indicates that it is the last line that was executed last and which caused the error.

However the "dominoes" do indeed start from the top, as that is where the execution, that ultimately led to the error, started from.

> Other than that, allow me a small rant for the addon that fails to launch.

Let me offer this alternative perspective - someone giving you something to use for free, does not entitle you to anything. Nor does offering support at one time, again for free, mean that you can expect that to continue forever.

Life is too short to get hung up on these things.

> That is why I'd rather learn python

That is a fine goal, but the sun has almost completely set on Python 2, and with regards to your actual issue, not only are you not helping yourself by not providing a log, by focusing on just the lines you have posted you are likely to go round in circles trying to find the problem.

If you do manage to figure things out, open a new issue specific to that problem. Let's end this one here.

**Comment by pitsi** [comment]
> Let me offer this alternative perspective - someone giving you something to use for free, does not entitle you to anything. Nor does offering support at one time, again for free, mean that you can expect that to continue forever.

I agree with what you say here, but...
I also spend time and effort to file an issue. And not a "addon not working, what do" issue, a detailed one with logs etc. And what I receive in return is some ungrateful reply by some fanboy who has never opened or helped on any issue, ever. When I proved him completely wrong in front of other users, he simply deleted his comments. And in return, I deleted every single issue I had ever opened in that dev's projects. They were not much, but if he and his fanboys do not care, so why should I? At this point, it is no longer frustration, it's plain anger.
They can rely on the forementioned users and their pathetic issue reports to fix any issues the addon has. I could continue for many more lines and post examples, but yea, lets end it here.
After all, as a debian user, I am used to maintainers packaging something and not updating it or replying to bug reports ever.

Ontopic now, here is a complete log of the error that you asked. I just altered any filenames that can give away the addon.
Tbh, I don't think creating a new issue for youtube would help solving it.
```
2024-05-24 17:05:29.573 T:140161581782784   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.UnicodeDecodeError'>
                                            Error Contents: 'ascii' codec can't decode byte 0xce in position 0: ordinal not in range(128)
                                            Traceback (most recent call last):
                                              File "/storage/.kodi/addons/plugin.video.pitsi/addon.py", line 14, in <module>
                                                from resources.lib import pitsi
                                              File "/storage/.kodi/addons/plugin.video.pitsi/resources/lib/pitsi.py", line 17, in <module>
                                                from youtube_resolver import resolve as yt_resolver
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_resolver.py", line 14, in <module>
                                                from youtube_plugin.youtube.provider import Provider
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/__init__.py", line 13, in <module>
                                                from .provider import Provider
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/provider.py", line 17, in <module>
                                                from .client import YouTube
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/client/__init__.py", line 13, in <module>
                                                from .youtube import YouTube
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/client/youtube.py", line 20, in <module>
                                                from .login_client import LoginClient
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/client/login_client.py", line 15, in <module>
                                                from .__config__ import (
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/client/__config__.py", line 195, in <module>
                                                _api_check = APICheck(XbmcContext())
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/context/xbmc/xbmc_context.py", line 321, in __init__
                                                self.parse_params(dict(parse_qsl(params)))
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/context/abstract_context.py", line 297, in parse_params
                                                param, value
                                            UnicodeDecodeError: 'ascii' codec can't decode byte 0xce in position 0: ordinal not in range(128)
                                            -->End of Python script error report<--

```

**Comment by MoojMidge** [comment]
> Ontopic now, here is a complete log of the error that you asked.

That is not what I asked for. I told you that nothing can be done without a full log and that concentrating on the error itself would not tell you anything. But this has gone on long enough and wasted enough time. I won't be responding to issues like this anymore.

**Comment by pitsi** [comment]
Unless the log I have to (or can) give is from a fresh installation of kodi and contains nothing, I won't provide a full log, ever.
As for the log above, the errors are the same even with debug enabled! Good luck to me :D

**Comment by pitsi** [comment]
After 2 days of testing, I have concluded that the error above is directly related to python2 and the youtube addon. 

My test machines were the one with kodi 18 I mentioned above (le 9.2 x64) and my main pc running kodi 21 (debian testing/unstable x64). I installed youtube's 7.0.5 (latest available version on the main repo), 7.0.6 (latest stable version here during the weekend) and 7.0.7-beta3 (latest beta version here during the weekend), plus the 2 latest versions of the addon which does not work due to the error. On all tests, every old version was uninstalled before installing the newer one and the addon's settings were also removed upon uninstall.
Every combination of youtube + other addon worked with zero issues on kodi 21 and python3, but the same combinations poped the same error as above on python2.
